src/components/RoomNameInput.js

```diff
@@ -82,45 +90,52 @@ class RoomNameInput extends Component {
         }
     }
 
-    /**
-     * Modifies the room name to follow our conventions:
-     * - Max length 80 characters
-     * - Cannot not include space or special characters, and we automatically apply an underscore for spaces
-     * - Must be lowercase
-     * Also checks to see if this room name already exists, and displays an error message if so.
-     * @param {String} roomName
-     *
-     */
-    checkAndModifyRoomName(roomName) {
-        const modifiedRoomNameWithoutHash = roomName.substring(1)
-            .replace(/ /g, '_')
-            .replace(/[^a-zA-Z\d_]/g, '')
-            .substring(0, CONST.REPORT.MAX_ROOM_NAME_LENGTH)
-            .toLowerCase();
-        const finalRoomName = `#${modifiedRoomNameWithoutHash}`;
-
+    validateRoomName(modifiedRoomName) {
         const isExistingRoomName = _.some(
             _.values(this.props.reports),
-            report => report && report.policyID === this.props.policyID && report.reportName === finalRoomName,
+            report => report && report.policyID === this.props.policyID && report.reportName === modifiedRoomName,
         );
 
         let error = '';
 
+
+        // We error if the user doesn't enter a room name or left blank
+        if (modifiedRoomName === CONST.POLICY.ROOM_PREFIX) {
+            error = this.props.translate('newRoomPage.pleaseEnterRoomName');
+        }
+
```

Comment:
I think we don't need this error here. onchange validation are added to RoomNameInput for Room Settings page where we expect to show the error as user types. On that page, save button is disabled when there is no input to roominput. Thus this error does not make sense there. 

Also,  as we are separately checking on WorkspaceNewRoompage.